# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  46773c43323e1f07b1a0bf421234c5173c585d94
NNN31 - Don't throw away previous activity when it reported non-zero downloads.

diff --git a/mail/components/activity/modules/pop3Download.sys.mjs b/mail/components/activity/modules/pop3Download.sys.mjs
--- a/mail/components/activity/modules/pop3Download.sys.mjs
+++ b/mail/components/activity/modules/pop3Download.sys.mjs
@@ -97,9 +97,9 @@ export var pop3DownloadModule = {
     // It can happen that download never started (e.g. couldn't connect to server),
     // with onDownloadStarted, but we still get a onDownloadCompleted event
     // when the connection is given up.
     const recentActivity = this._mostRecentActivityForFolder.get(aFolder.URI);
-    if (recentActivity) {
+    if (recentActivity && !recentActivity.numMsgsDownloaded) {
       this.activityMgr.removeActivity(recentActivity.eventID);
     }
 
     let displayText;
